### PR TITLE
Add issue form templates

### PR DESCRIPTION
**Pull request by timja**
Inspired by @NotMyFault's port, I did minor changes on top to the bug report to improve formatting of the environment section

see https://github.com/jenkinsci/.github/pull/51#issuecomment-1001070142

Example issus:
https://github.com/timja/.github/issues

Main one being the bug report:
https://github.com/timja/.github/issues/2

CI build fixed in https://github.com/jenkinsci/.github/pull/63

### CODE REVIEW

**Review by NotMyFault — CHANGES_REQUESTED** [review]
Otherwise the issue templates looking good as-is

**Review by NotMyFault — APPROVED** [review]

**Comment by timja** [comment]
@MarkEWaite you able to take a look?

**Review by MarkEWaite — APPROVED** [review]
The change looks great to me.  Does the CI job need to pass before it is merged?

**Comment by timja** [comment]
> The change looks great to me. Does the CI job need to pass before it is merged?

Just needs a merge with master probably I fixed it separately

File: .github/ISSUE_TEMPLATE/1-report-bug.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        2. Step 2: ...
+    validations:
+      required: true
+
+  - type: textarea
+    attributes:
+      label: Expected Results
+      description: What was your expected result?
+    validations:
+      required: true
```
Comment (jglick) [inline .github/ISSUE_TEMPLATE/1-report-bug.yml:82]: This is now making it impossible to enter a freeform bug report. Can we just have a template _Other_ that gives you a blank space?

Comment (jglick) [reply .github/ISSUE_TEMPLATE/1-report-bug.yml:82]: Workaround: file a **Documentation** issue, then clean up the **Description** after filed and correct the labels. https://github.com/jenkinsci/incrementals-tools/issues/21#event-5821597462

Comment (NotMyFault) [reply .github/ISSUE_TEMPLATE/1-report-bug.yml:82]: Not _completely_ impossible, the cli does not know about templates 👀 

But I get your point, for quick notifications or follow ups forms may be unnecessary

Comment (jglick) [reply .github/ISSUE_TEMPLATE/1-report-bug.yml:82]: The form is nice to guide new contributors who may not be used to thinking in terms of the information a maintainer would need to evaluate something filed out of the blue. It gets in the way for maintainers filing issues for themselves, and experienced contributors with an established working relationship who would already understand what is critical to include in an issue.

Comment (NotMyFault) [reply .github/ISSUE_TEMPLATE/1-report-bug.yml:82]: I guess flipping the boolean switch in the config would be sufficient. It doesn't add a separate button for blank issues but the following entry at the bottom of the templates table:

Comment (timja) [reply .github/ISSUE_TEMPLATE/1-report-bug.yml:82]: continued in https://github.com/jenkinsci/.github/pull/64

**Pull request merged**